store/tikv/2pc.go

```diff
@@ -831,15 +864,75 @@ type RelatedSchemaChange struct {
 	Amendable        bool
 }
 
-func (c *twoPhaseCommitter) checkSchemaValid() error {
+func (c *twoPhaseCommitter) tryAmendTxn(ctx context.Context, startInfoSchema SchemaVer, change *RelatedSchemaChange) (bool, error) {
+	memAmended := false
+	addMutations, _, err := c.txn.schemaAmender.AmendTxn(ctx, startInfoSchema, change, c.mutations)
+	if err != nil {
+		return false, err
+	}
+	// Prewrite new mutations.
+	if addMutations != nil && len(addMutations.keys) > 0 {
+		prewriteBo := NewBackofferWithVars(ctx, PrewriteMaxBackoff, c.txn.vars)
+		err = c.prewriteMutations(prewriteBo, *addMutations)
```

Comment:
Will this violate the isolation? Before retry, the result has been returned to the client already.

Reply:
The transaction is still in the `prewrite` phase, new `commitTS` will be fetched and after that primary key will be committed, and then the result will be returned to the client.